llama_index/objects/base.py

```diff
@@ -64,3 +67,52 @@ def as_retriever(self, **kwargs: Any) -> ObjectRetriever:
 
     def as_node_retriever(self, **kwargs: Any) -> BaseRetriever:
         return self._index.as_retriever(**kwargs)
+
+    def persist(
+        self,
+        persist_dir: str = DEFAULT_PERSIST_DIR,
+    ) -> None:
+        # try to persist object node mapping
+        try:
+            self._object_node_mapping.persist(persist_dir=persist_dir)
+        except (NotImplementedError, pickle.PickleError) as err:
+            warnings.warn(
+                (
+                    "Unable to persist ObjectNodeMapping. You will need to "
+                    "reconstruct the same object node mapping to build this ObjectIndex"
+                ),
+                DeprecationWarning,
+            )
+        self._index._storage_context.persist(persist_dir=persist_dir)
+
+    @classmethod
+    def from_persist_dir(
+        cls,
+        persist_dir: str = DEFAULT_PERSIST_DIR,
+        object_node_mapping: Optional[BaseObjectNodeMapping] = None,
+        object_node_mapping_type: Optional[Type[BaseObjectNodeMapping]] = None,
```

Comment:
hmm, it feels a little weird to have to pass the type here -- Do you think it's better to just have all the pickling logic in this function?

Reply:
Yeah, I didn't like this too much either. Okay, let me see if there's something that can be a bit less strange.